<?xml version="1.0"?>

<launch>

	<include file="$(find robotino_calibration)/ros/launch/pi_tag/fiducials.launch"/>

	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_robot2camera" output="screen" args="-0.01957318079 -0.20404139431 0.0 0.0 1.57079632679 0.39269908169 torso_knob_link ensenso_frame 100"/>

	<!-- transform between marker (e.g. tag or checkerboard) coordinate system and landmark_reference coordinate system
		tag_x: for a PiTag with ID "x" - starts in upper left corner of Pi tag pattern, x goes horizontal to the right, y goes downwards to gravity direction, z goes into the plane
		landmark_reference: similar definition as tag_x, but located at the height of the laser scanner on the left edge of the box in front of the wall or in the corner of two walls
		transform between them: no rotation, just an x, y, and z offset (ATTENTION: height (y or z) offset depends on mounting height of the laser scanner) -->

	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_73" output="screen" args="0 1.274 0.617 0 0 0  landmark_reference_nav tag_73 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_79" output="screen" args="-0.142 0.876 0.03 0 0 0  landmark_reference_nav tag_79 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_69" output="screen" args="0 0.684 1.782 0 0 0  landmark_reference_nav tag_69 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_25" output="screen" args="-0.563 0 0.522 0 0 0  landmark_reference_nav tag_25 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_55" output="screen" args="-0.143 0.565 0.03 0 0 0  landmark_reference_nav tag_55 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_38" output="screen" args="0 1.534 1.458 0 0 0  landmark_reference_nav tag_38 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_36" output="screen" args="-0.265 0 0.774 0 0 0  landmark_reference_nav tag_36 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_64" output="screen" args="-0.138 1.166 0.03 0 0 0  landmark_reference_nav tag_64 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_landmark_tag_48" output="screen" args="0 0.684 1.018 0 0 0  landmark_reference_nav tag_48 100"/>
	
	<node ns="camera_base_calibration_pitag" name="camera_base_calibration" pkg="robotino_calibration" type="camera_base_calibration" output="screen">
		<rosparam command="load" file="$(find robotino_calibration)/ros/launch/camera_base_calibration_pitag_params_RAW.yaml"/>
		<param name="marker_type" value="pitag" />
		<remap from="~get_fiducials" to="/fiducials/get_fiducials"/>
	</node>
</launch>
